### PR TITLE
GEN 1184 - Added Workflow Classification and Metric LevelConfig

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes:
- Added Workflow Classification and Metric LevelConfig
- Refactored metadata source to accept entityReference (and not just Table)

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

New feature
- [x] The issue properly describes why the new feature is needed, what's the goal, and how we are building it. Any discussion
    or decision-making process is reflected in the issue.
- [x] I have updated the documentation.
- [x] I have added tests around the new logic.

### CODE REVIEW

File: ingestion/src/metadata/profiler/source/fetcher/fetcher_strategy.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+                or not self._filter_schemas(table)
+            )
+            and (
+                not self.source_config.tableFilterPattern
+                or not self._filter_tables(table)
+            )
+        ]
+
+        if self.source_config.classificationFilterPattern:
+            tables = self.filter_classifications(tables)  # type: ignore
```
Comment (pmbrull) [inline ingestion/src/metadata/profiler/source/fetcher/fetcher_strategy.py:252]: I see the `filter_classifications` eats an `Iterable`. Would it make sense to have the `_filter_*` methods to work on individual assets? Might make it easier to reuse and chain filters in a single comprehension, something like:

```
[
            table
            for table in tables
            if (
                not self.source_config.schemaFilterPattern
                or not self._filter_schemas(table)
            )
            and (
                not self.source_config.tableFilterPattern
                or not self._filter_tables(table)
            )
            and  not self._filter_classification(table)
        ]
```

Comment (TeddyCr) [reply ingestion/src/metadata/profiler/source/fetcher/fetcher_strategy.py:252]: Yep, that makes sense. I started with the iterable and then made schema and table eat an entity. Let me update that.

**Review by pmbrull — APPROVED** [review]

File: ingestion/tests/integration/postgres/test_data_quality.py [hunk 2]
```diff
@@ -217,65 +213,6 @@ def run_data_quality_workflow(
         metadata.delete(TestSuite, test_suite.id, recursive=True, hard_delete=True)
 
 
-def test_all_definition_exists(metadata, run_data_quality_workflow, db_service):
```
Comment (sushi30) [inline ingestion/tests/integration/postgres/test_data_quality.py:220]: Note that this tests coverage on test cases in integration tests. It achieves something different than the test implemented in this PR.

File: ingestion/src/metadata/profiler/source/fetcher/entity_fecther.py
Comment (sushi30) [inline ingestion/src/metadata/profiler/source/fetcher/entity_fecther.py:1]: typo in file name

File: ingestion/src/metadata/profiler/source/fetcher/fetcher_strategy.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    filter_by_database,
+    filter_by_schema,
+    filter_by_table,
+)
+
+FIELDS = ["tableProfilerConfig", "columns", "customMetrics", "tags"]
+
+
+class FetcherStrategy(ABC):
+    """Fetcher strategy interface"""
```
Comment (sushi30) [inline ingestion/src/metadata/profiler/source/fetcher/fetcher_strategy.py:50]: Can you add a bit more information about this class? When would we want to use it and in what context?

File: ingestion/tests/integration/postgres/test_data_quality.py (hunk 2 above)
Comment (TeddyCr) [reply ingestion/tests/integration/postgres/test_data_quality.py:220]: Can you share more details?

**Review by pmbrull — APPROVED** [review]

**Review by pmbrull — APPROVED** [review]

**Pull request merged**